test/e2e_node/services/server.go

```diff
@@ -282,34 +283,28 @@ func (s *server) kill() error {
 	}
 
 	// Attempt to shut down the process in a friendly manner before forcing it.
-	waitChan := make(chan error)
 	go func() {
 		_, err := cmd.Process.Wait()
-		waitChan <- err
-		close(waitChan)
 	}()
 
 	const timeout = 10 * time.Second
 	for _, signal := range []string{"-TERM", "-KILL"} {
 		klog.V(2).Infof("Killing process %d (%s) with %s", pid, name, signal)
-		cmd := exec.Command("kill", signal, strconv.Itoa(pid))
+		ctx, cancel := context.WithTimeout(context.Background(), timeout)
+		defer cancel()
+		cmd := exec.CommandContext(ctx, "kill", signal, strconv.Itoa(pid))
 		_, err := cmd.Output()
```

Comment:
Maybe switch to Run, which doesn't return extra parameters.


err := exec.CommandContext(ctx, "kill", signal, strconv.Itoa(pid)).Run()

...but actually writing this comment made me try it, and both Run and Output don't actually seem to work: https://play.golang.org/p/byUW9mLoqAX

(in that it does not actually return early when the context times out.)

Reply:
I tried it on my local and seems to work.
`context error context deadline exceeded; duration 1.350341ms`

It somehow doesn't work on the play.golang. I tried the earlier example of the code that I pasted and it had the same behavior on play.golang but works on my local. Quite strange.

System config:
```
go version go1.14 linux/amd64
Linux linux 5.3.0-53-generic #47~18.04.1-Ubuntu SMP Thu May 7 13:10:50 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```